app/services/purchase/create_service.rb

```diff
@@ -242,6 +242,9 @@ def build_purchase(params_for_purchase)
       purchase.seller = product.user
       purchase.is_gift_sender_purchase = is_gift? unless params_for_purchase.has_key?(:is_gift_receiver_purchase)
       purchase.offer_code = product.find_offer_code(code: purchase.discount_code.downcase.strip) if purchase.discount_code.present?
+      if purchase.offer_code.present? && product.default_discount_code.present? && purchase.offer_code.id == product.default_discount_code.id
+        purchase.used_default_discount_code = true
```

Comment:
```suggestion
        purchase.default_offer_code_id = purchase.offer_code.id
```

We want to know the exact offer code used.